sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -39,6 +40,9 @@ abstract class FileTable(
     userSpecifiedSchema: Option[StructType])
   extends Table with SupportsRead with SupportsWrite {
 
+  // If `partitioning` contains nested columns, an `AnalysisException` will be thrown
+  partitioning.toSeq.validatePartitionColumns()
```

Comment:
The `partitioning` here is always inferred, so this check always passes.

Reply:
Yeah seems like so. Even after we support passing the partitioning from `TableProvider`, the partitioning columns should be validated in Catalog already.